packages/flutter/lib/src/material/spell_check_suggestions_toolbar.dart

```diff
@@ -0,0 +1,218 @@
+
+  /// The buttons that will be displayed in the spell check suggestions toolbar.
+  final List<ContextMenuButtonItem> buttonItems;
+
+  /// Builds the default Android Material spell check suggestions toolbar.
+  static Widget _spellCheckSuggestionsToolbarBuilder(BuildContext context, Widget child) {
+    return _SpellCheckSuggestionsToolbarContainer(
+      child: child,
+    );
+  }
+
+  /// Builds the button items for the toolbar based on the available
+  /// spell check suggestions.
+  static List<ContextMenuButtonItem>? buildButtonItems(
+    BuildContext context,
+    EditableTextState editableTextState,
+  ) {
+    // Determine if composing region is misspelled.
+    final SuggestionSpan? spanAtCursorIndex =
+      findSuggestionSpanAtCursorIndex(
+        editableTextState.currentTextEditingValue.selection.baseOffset,
+        editableTextState.spellCheckResults!.suggestionSpans,
+    );
+
+    if (spanAtCursorIndex == null) {
+      return null;
+    }
+
+    final List<ContextMenuButtonItem> buttonItems = <ContextMenuButtonItem>[];
+
+    // Build suggestion buttons.
+    for (final String suggestion in spanAtCursorIndex.suggestions) {
+      buttonItems.add(ContextMenuButtonItem(
+        onPressed: () {
+          editableTextState
+            .replaceComposingRegion(
+              SelectionChangedCause.toolbar,
+              suggestion,
+          );
+        },
+        label: suggestion,
+      ));
+    }
+
+    // Build delete button.
+    final ContextMenuButtonItem deleteButton =
+      ContextMenuButtonItem(
+        onPressed: () {
+          editableTextState.replaceComposingRegion(
+            SelectionChangedCause.toolbar,
+            '',
+          );
+        },
+        type: ContextMenuButtonType.delete,
+    );
+    buttonItems.add(deleteButton);
+
+    return buttonItems;
+  }
+
+  /// Determines the Offset that the toolbar will be anchored to.
+  static Offset getToolbarAnchor(TextSelectionToolbarAnchors anchors) {
+    return anchors.secondaryAnchor == null ? anchors.primaryAnchor : anchors.secondaryAnchor!;
+  }
+
+  /// Builds the toolbar buttons based on the [buttonItems].
+  List<Widget> _buildToolbarButtons(BuildContext context) {
+    return buttonItems.map((ContextMenuButtonItem buttonItem) {
+      final TextSelectionToolbarTextButton button =
+        TextSelectionToolbarTextButton(
+          padding: const EdgeInsets.fromLTRB(20, 0, 0, 0),
+          onPressed: buttonItem.onPressed,
+          alignment: Alignment.centerLeft,
+          child: Text(AdaptiveTextSelectionToolbar.getButtonLabel(context, buttonItem)),
+        );
+
+      if (buttonItem.type == ContextMenuButtonType.delete) {
+        return DecoratedBox(
+          decoration: const BoxDecoration(border: Border(top: BorderSide(color: Colors.grey))),
+          child: button.copyWith(
+            child: Text(AdaptiveTextSelectionToolbar.getButtonLabel(context, buttonItem), style: const TextStyle(color: Colors.blue)),
```

Comment:
Nit: Is it any cleaner to do it more like this instead of the copyWith?

```dart
      final TextSelectionToolbarTextButton button =
        TextSelectionToolbarTextButton(
          padding: const EdgeInsets.fromLTRB(20, 0, 0, 0),
          onPressed: buttonItem.onPressed,
          alignment: Alignment.centerLeft,
          style: buttonItem.type == ContextMenuButtonType.delete ? const TextStyle(color: Colors.blue) : null,
          child: Text(AdaptiveTextSelectionToolbar.getButtonLabel(context, buttonItem)),
        );

        if (buttonItem.type != ContextMenuButtonType.delete) {
          return button;
        }
        return DecoratedBox(
          decoration: const BoxDecoration(border: Border(top: BorderSide(color: Colors.grey))),
          child: button,
        );
```